client/galaxy/scripts/mvc/library/library-foldertoolbar-view.js

```diff
@@ -1387,9 +1394,25 @@ var FolderToolbarView = Backbone.View.extend({
     '<strong>Choose the datasets to import:</strong>',
     '<ul>',
       '<% _.each(history_contents, function(history_item) { %>',
-        '<li data-id="<%= _.escape(history_item.get("id")) %>">',
-          '<input style="margin: 0;" type="checkbox"> <%= _.escape(history_item.get("hid")) %>: <%= _.escape(history_item.get("name")) %>',
-        '</li>',
+        '<% if (history_item.get("deleted") != true ) { %>',
+          '<% if (history_item.get("type") === "collection") { %>',
+              '<% var collection_type = history_item.get("collection_type") %>',
+              '<% if (collection_type === "list") { %>',
+                '<li data-id="<%= _.escape(history_item.get("id")) %>" data-name="<%= _.escape(history_item.get("type")) %>">',
+                  '<input style="margin: 0;" type="checkbox"> <%= _.escape(history_item.get("hid")) %>: <%= _.escape(history_item.get("name")) %> (Dataset Collection)',
+                '</li>',
+               '<% } else { %>',
+                 '<li><input style="margin: 0;" type="checkbox" onclick="return false;" disabled="disabled">',
+                   '<a title="You can convert this collection into a collection of type list using the Collection Tools" href="" onclick="return false;" style="text-decoration: none">',
+                     '<%= _.escape(history_item.get("hid")) %>: <%= _.escape(history_item.get("name")) %> (Dataset Collection of type <%= _.escape(collection_type) %> not supported.)',
+                  '</a></li>',
+                '<% } %>',
+          '<% } else if (history_item.get("visible") === true && history_item.get("state") === "ok") { %>',
```

Comment:
@martenson filtering collection elements is happening here. I think it is more logical that you can only add datasets that are visible and whose state is ok. Otherwise in a history with large collections you'd have a huge amount of datasets to select. But I can certainly revert that if you prefer.